uint8 prismType
---
uint16 rc